### ISSUE TITLE
(gen_img[:, 0, 0, 0] * 0 + loss_Gpl) ... but why?

### ISSUE THREAD

**Issue by Mwni** [NVlabs/stylegan2-ada-pytorch#163]
In `loss.py`, why is this being done?

```
(gen_img[:, 0, 0, 0] * 0 + loss_Gpl).mean().mul(gain).backward()
```
and
```
(real_logits * 0 + loss_Dreal + loss_Dr1).mean().mul(gain).backward()
```
both times, a term multiplied by zero.

What did they mean by this?

**Comment by BenQLange** [comment]
Anyone knows?

**Comment by Hong753** [comment]
I found that without adding the zero tensors, they don't work with DDP due to some issue with torch.autograd.grad
